### ISSUE TITLE
Ошибка iptables в add100redirect.sh

### ISSUE THREAD

**Issue by Ka6an4eG** [tas-unn/bypass_keenetic#8]
Из-за этой ошибки обход через сторонний vpn не работает.

~ # iptables -I PREROUTING -w -t nat -i br0 -p tcp -m set --match-set unblockvpn dst -j MASQUERADE -o ppp1

iptables v1.4.21: Can't use -o with PREROUTING

Try `iptables -h' or 'iptables --help' for more information.

**Comment by tas-unn** [comment]
Да, к сожалению я знаю об этой проблеме. Не могу пока её решить... по идее лучше спрашивать вот в этой теме
https://forum.keenetic.com/topic/8106-%D0%B2%D1%8B%D0%B1%D0%BE%D1%80%D0%BE%D1%87%D0%BD%D1%8B%D0%B9-%D1%80%D0%BE%D1%83%D1%82%D0%B8%D0%BD%D0%B3-%D1%87%D0%B5%D1%80%D0%B5%D0%B7-vpn-%D1%82%D1%83%D0%BD%D0%BD%D0%B5%D0%BB%D1%8C/

**Comment by ziwork** [comment]
@Ka6an4eG @tas-unn 
Привет.

Решил вопрос с заворачиванием в vpn. 
Скоро выпущу в релиз. Осталось причесать код, поправить бота.
Если успею, то обновление ждите сегодня.

**Comment by ziwork** [comment]
> @ziwork
> 
> Отлично

@Ka6an4eG проверяйте, пользуйтесь. :-)

**Comment by Ka6an4eG** [comment]
> > @ziwork
> > Отлично
> 
> @Ka6an4eG проверяйте, пользуйтесь. :-)

Не работает. Я не совсем понял, в каком месте теперь указывается какой интерфейс указывать как vpn-подключение? Раньше это прописывалось в add100redirect.sh

**Comment by tas-unn** [comment]
> > > @ziwork
> > > Отлично
> > 
> > 
> > @Ka6an4eG проверяйте, пользуйтесь. :-)
> 
> Не работает. Я не совсем понял, в каком месте теперь указывается какой интерфейс использовать как vpn-подключение? Раньше это прописывалось в add100redirect.sh

мы работаем над этим. В будущем этого файла, скорее всего, не будет

**Comment by Ka6an4eG** [comment]
> > > > @ziwork
> > > > Отлично
> > > 
> > > 
> > > @Ka6an4eG проверяйте, пользуйтесь. :-)
> > 
> > 
> > Не работает. Я не совсем понял, в каком месте теперь указывается какой интерфейс использовать как vpn-подключение? Раньше это прописывалось в add100redirect.sh
> 
> мы работаем над этим. В будущем этого файла, скорее всего, не будет

Так а что проверять-то тогда? 🤷‍♀️ Меня маршрутизация через собственный vpn

**Comment by ziwork** [comment]
> > > > > @ziwork
> 
> > > > > Отлично
> 
> > > > 
> 
> > > > 
> 
> > > > @Ka6an4eG проверяйте, пользуйтесь. :-)
> 
> > > 
> 
> > > 
> 
> > > Не работает. Я не совсем понял, в каком месте теперь указывается какой интерфейс использовать как vpn-подключение? Раньше это прописывалось в add100redirect.sh
> 
> > 
> 
> > мы работаем над этим. В будущем этого файла, скорее всего, не будет
> 
> 
> 
> Так а что проверять-то тогда? 🤷‍♀️ Меня интересует маршрутизация через собственный vpn 

Сейчас реализовано так:
Скрипт должен определить какой впн подключён и под него создать правила. В боте будет возможность добавить адреса. Добавить в список, vpn, добавить. По идее должно все заработать. Ещё вопрос, свой впн это какой? Я проверял на SSTP. Сейчас работаем, что бы на каждый vpn, был отдельный файл с адресами. Пока реализовано что список один для любого vpn.

**Comment by Ka6an4eG** [comment]
> > > > > > @ziwork
> > 
> > 
> > > > > > Отлично
> > 
> > 
> > > > >
> > 
> > 
> > > > >
> > 
> > 
> > > > > @Ka6an4eG проверяйте, пользуйтесь. :-)
> > 
> > 
> > > >
> > 
> > 
> > > >
> > 
> > 
> > > > Не работает. Я не совсем понял, в каком месте теперь указывается какой интерфейс использовать как vpn-подключение? Раньше это прописывалось в add100redirect.sh
> > 
> > 
> > > 
> > 
> > 
> > > мы работаем над этим. В будущем этого файла, скорее всего, не будет
> > 
> > 
> > Так а что проверять-то тогда? 🤷‍♀️ Меня интересует маршрутизация через собственный vpn
> 
> Сейчас реализовано так: Скрипт должен определить какой впн подключён и под него создать правила. В боте будет возможность добавить адреса. Добавить в список, vpn, добавить. По идее должно все заработать. Ещё вопрос, свой впн это какой? Я проверял на SSTP. Сейчас работаем, что бы на каждый vpn, был отдельный файл с адресами. Пока реализовано что список один для любого vpn.

У меня wireguard. Как я могу проверить, что правила для впн были созданы?

**Comment by ziwork** [comment]
> > > > > > > @ziwork
> > > 
> > > 
> > > > > > > Отлично
> > > 
> > > 
> > > > > >
> > > 
> > > 
> > > > > >
> > > 
> > > 
> > > > > > @Ka6an4eG проверяйте, пользуйтесь. :-)
> > > 
> > > 
> > > > >
> > > 
> > > 
> > > > >
> > > 
> > > 
> > > > > Не работает. Я не совсем понял, в каком месте теперь указывается какой интерфейс использовать как vpn-подключение? Раньше это прописывалось в add100redirect.sh
> > > 
> > > 
> > > > 
> > > 
> > > 
> > > > мы работаем над этим. В будущем этого файла, скорее всего, не будет
> > > 
> > > 
> > > Так а что проверять-то тогда? 🤷‍♀️ Меня интересует маршрутизация через собственный vpn
> > 
> > 
> > Сейчас реализовано так: Скрипт должен определить какой впн подключён и под него создать правила. В боте будет возможность добавить адреса. Добавить в список, vpn, добавить. По идее должно все заработать. Ещё вопрос, свой впн это какой? Я проверял на SSTP. Сейчас работаем, что бы на каждый vpn, был отдельный файл с адресами. Пока реализовано что список один для любого vpn.
> 
> У меня wireguard. Как я могу проверить, что правила для впн были созданы?

Проверить можно так:
ip route show table 1001
ip rule
ipset list unblockvpn
iptables-save | grep 0xd1000

**Comment by Ka6an4eG** [comment]
@ziwork 
~ # ip route show table 1001
~ # ip rule
0:      from all lookup local
10:     from all fwmark 0xffffcff lookup main
196:    from 172.20.67.18 lookup 58
197:    from 10.66.66.7 lookup 59
32766:  from all lookup main
32767:  from all lookup default
~ # ipset list unblockvpn
Name: unblockvpn
Type: hash:net
Revision: 6
Header: family inet hashsize 1024 maxelem 65536
Size in memory: 1212
References: 0
Members:
188.114.97.1
31.13.72.6
104.21.84.156
104.16.155.36
172.67.194.145
188.114.99.234
104.16.154.36
172.67.74.128
157.240.194.63
188.114.98.234
104.26.5.133
104.26.4.133
157.240.194.174
188.114.96.1
31.13.72.53
31.13.72.174
~ # iptables-save | grep 0xd1000

Строка 197:    from 10.66.66.7 lookup 59 судя по IP, относится к моему vpn

**Comment by ziwork** [comment]
@Ka6an4eG Привет. 
Выпустить доработанную версию. Можете попробовать.
https://github.com/ziwork/bypass_keenetic

**Comment by Ka6an4eG** [comment]
> @Ka6an4eG Привет. Выпустить доработанную версию. Можете попробовать. https://github.com/ziwork/bypass_keenetic

После нажатия DNS Override ВКЛ и перезагрузки роутера сайты вообще перестают открываться
